### PR TITLE
396 update anvil catalog entity list config

### LINKED ISSUE #396: Update anvil-catalog entity list config
#### Need
We are starting to refine entity list configurations into a single style for both API and TSV-driven sites. The `anvil-catalog` entity list configuration is currently in a deprecated format and needs to be updated to match the current preferred paradigm. Specifically, we need to update the workspace entity list configuration.

#### Approach
* Follow the approach/spec described in the [configuration guide](https://github.com/clevercanary/data-browser/blob/main/docs/configuration-guide.md) for configuring entity lists.
* Use the `ncpi-catalog` `studiesEntityConfig.ts` and its corresponding view builder and transformer as an example.
* Replace any reference to the Text component to use the Cell component instead.
* Any sanitizing of the entity value should occur in the transformer.
* Ensure transformer functions return either a string or number value (and not null or undefined).

#### Notes
* The existing `AnvilSourceItem` response model can be reused as is but needs to be moved to the appropriate location.
* The existing `workspaceTransformer.ts` is currently a mix of a view model builder and transformer functionality. Split/move/update the functionality as needed then remove this file.

#### Definition of Done
* Workspace entity config is updated to match spec.
* `anvil-catalog`-specific view builders file created.
* `anvil-catalog`-specific transformers file created (with the existing `workspaceTransformer.ts` removed).

### PR DESCRIPTION
**Pull request by jpaten**
### Ticket
Closes #396.

### Reviewers
@MillenniumFalconMechanic .

### Changes
- Separated viewbuilders & transformers and moved to correct folders as specified by the configuration guide
- Moved `anvil-catalog` & `anvil` model builders to correct folders as specified by the configuration guide
- Fixed minor typo (namving -> naming) in configuration guide

### Definition of Done (from ticket)
  - Workspace entity config is updated to match 
  - anvil-catalog-specific view builders file created.
  - anvil-catalog-specific transformers file created (with the existing workspaceTransformer.ts removed).

### Known Issues
- Kept currently unused `anvil-catalog` model builder

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -13,7 +13,7 @@
       the copyright owner that is granting the License.
 
       "Legal Entity" shall mean the union of the acting entity and all
-      other entities that control, are controlled by, or are under common
+      other workspaceEntityConfig that control, are controlled by, or are under common
```
Comment (MillenniumFalconMechanic) [inline LICENSE:16]: Looks like you did a rename that went a little wide! :) Would you be able to revert this change?

File: docs/configuration-guide.md [hunk 2]
```diff
@@ -5,14 +5,14 @@ Entity lists display result sets from either an API endpoint or a TSV and can be
```
Comment (MillenniumFalconMechanic) [inline docs/configuration-guide.md:5]: Same as above!

File: explorer/app/apis/azul/common/entities.ts [hunk 3]
```diff
@@ -10,14 +10,14 @@ import { FileEntityResponse } from "../anvil/common/entities";
 import { CategoryKey, CategoryValueKey } from "../../../common/entities";
```
Comment (MillenniumFalconMechanic) [inline explorer/app/apis/azul/common/entities.ts:10]: Revert renaming!

**Comment by jpaten** [comment]
Looks like I might also have not separated the model builders and transformers properly. In this configuration it appears that the model builders add text formatting, but based on the guide and the hca-dcp implementation (#401) that should probably happen in the transformers. Let me know which way around is right(or if it doesn't matter) and I'll change if necessary!

**Comment by jpaten** [comment]
Wait nvm think I misunderstood disregard the prior comment

File: docs/configuration-guide.md [hunk 4]
```diff
@@ -5,7 +5,7 @@ Entity lists display result sets from either an API endpoint or a TSV and can be
 
 #### 1. Add Response Model
 Add an interface matching the response model expected from the API endpoint or TSV row.
-* Namving convention: `entities.ts`.
+* Naming convention: `entities.ts`.
```
Comment (MillenniumFalconMechanic) [inline docs/configuration-guide.md:8]: Thank you!

File: explorer/app/viewModelBuilders/anvil-catalog/common/viewModelBuilders.ts [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+});
+
+/**
+ * Build props for study name cell component from the given study in the Anvil catalog.
+ * @param source - Study row in the Anvil catalog TSV.
+ * @returns Model to be used as props for the platform cell.
+ */
+export const buildStudyName = (
+  source: AnvilSourceItem
+): React.ComponentProps<typeof C.Text> => createColumn(getStudyName(source));
```
Comment (MillenniumFalconMechanic) [inline explorer/app/viewModelBuilders/anvil-catalog/common/viewModelBuilders.ts:28]: Replace usage of `Text` with `Cell`, here and throughout (and also in the view model builder). Ideally we'd just call `ViewModelBuilder.getStudyName()` here (and we can then also remove the `createColumn` function above).

File: explorer/app/apis/anvil-catalog/common/transformers.ts [hunk 6]
```diff
@@ -0,0 +1,21 @@
+import { AnvilSourceItem } from "../../anvil/common/entities";
+
+export const getStudyName = (source: AnvilSourceItem): string => source?.name;
```
Comment (MillenniumFalconMechanic) [inline explorer/app/apis/anvil-catalog/common/transformers.ts:3]: I think we can remove the optional chaining (`source?`) as the `source` param isn't optional.

Would you mind adding nullish coalescing (returning "") to each function here in the transformer so the transformer functions always return a string value (and not undefined or null).

**Review by MillenniumFalconMechanic — CHANGES_REQUESTED** [review]
Looking good Jonah, thank you! I have a couple of small nits for you to take a look at when you have a moment.

**Pull request closed without merging**